dli_app/mod_auth/controllers.py

```diff
@@ -103,3 +115,67 @@ def logout():
         logout_user()
     flash('You have successfully logged out.', 'alert-success')
     return redirect(url_for('default.home'))
+
+@mod_auth.route('/resetpass/', methods=['GET', 'POST'])
+def resetpass():
+    """Reset the user's password
+
+    If the user successfully submitted the form, send a password 
+    reset email. Otherwise, render the reset form again.
+    """
+
+    form = ForgotForm()
+    if form.validate_on_submit():
+        # Email is authenticated and sent
+        email = form.email.data
+	#Create User Specific URL here
+        pw_reset = PasswordReset(
+            user= User.get_by_email(email),
+            key=''.join(
+                random.choice(
+                    string.ascii_letters + string.digits
+                ) for _ in range(60)
+            ),
```

Comment:
Personal suggestion:
Change PasswordReset to set key=None as default, and within __init__, if key=None, generate it there.
That will make this section a bit less cluttered.

```python
def __init__(self, user, key=None):
    if key is None:
        key = random.choice(...etc...)
    self.user = user
    self.key = key
```